kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/spec/style/scopes/ContainerScope.kt

```diff
@@ -171,6 +173,36 @@ interface ContainerScope : TestScope {
       })
    }
 
+   fun before(f: Before) {
+      if (hasChildren()) throw outOfOrderCallbacksException
+      val thisTestCase = this.testCase
+      var hasRun = false
+
+      appendExtension(object : TestListener {
+         override suspend fun before(testCase: TestCase) {
+            if (!hasRun && thisTestCase.descriptor.isAncestorOf(testCase.descriptor)) {
```

Comment:
What's the purpose of the hasRun ?

Reply:
That was to prevent the `before` hook from running more than once.